server/services/passport.js

```diff
@@ -0,0 +1,67 @@
+const passport = require("passport");
+const GooleStrategy = require("passport-google-oauth20").Strategy;
+const FacebookStrategy = require("passport-facebook").Strategy;
+const keys = require("../config/keys");
+const mongoose = require("mongoose");
+const User = mongoose.model("users");
+
+passport.serializeUser((user, done) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser((id, done) => {
+  User.findById(id).then((user) => {
+    done(null, user);
+  });
+});
+
+passport.use(
+  new GoogleStrategy({
+      clientID: keys.googleClientID,
+      clientSecret: keys.googleClientSecret,
+      callbackURL: "/auth/google/callback",
+      proxy: true
+    },
+    (accessToken, refreshToken, profile, done) => {
+      User.findOne({ userId: profile.id })
+        .then((existingUser) => {
+          if (existingUser) {
+            done(null, existingUser);
+          } else {
+            new User({
+              userId: profile.id,
+              username: profile.displayName,
+              picture: profile._json.picture
+            }).save()
+              .then((user) => {
+                done(null, user);
+              });
+          }
+        });
+    }
+  )
+);
+
+passport.use(new GitHubStrategy({
+    clientID: keys.githubClientID,
+    clientSecret: keys.githubClientSecret,
+    callbackURL: "/auth/github/callback"
+  },
+  function(accessToken, refreshToken, profile, done) {
+    User.findOne({ userId: profile.id })
+      .then((existingUser) => {
+        if (existingUser) {
+          done(null, existingUser);
+        } else {
+          new User({
+            userId: profile.id,
```

Comment:
Also how we ensuring `profile.id` is same for github and google logins?
Here, `profile.id` is different for both github and google.

We need to prevent duplicate accounts here